src/components/players/list/index.js

```diff
@@ -11,33 +11,27 @@ import { formatPlayerPhotoUrl } from 'utils/stringUtils';
 
 import styles from './index.module.scss';
 
-class List extends React.Component {
-  constructor(props) {
-    super(props);
+const List = ({
+  player: {
+    details: { person_id: selectedPlayerId },
+  },
+  players,
+  selectedTeam,
+  selectedTeam: { teamId, teamColor, urlName },
+  history,
+  getSelectedPlayer,
+}) => {
+  const domRef = useRef();
+  const selectedTeamRef = useRef(selectedTeam);
 
-    this.ref = createRef();
-  }
-
-  componentDidUpdate(prevProps) {
-    const { selectedTeam } = this.props;
-    const { selectedTeam: selectedTeamPrev } = prevProps;
-
-    if (selectedTeamPrev !== selectedTeam) {
-      this.ref.current.scrollTo(0, 0);
+  useEffect(() => {
+    if (selectedTeamRef.current !== selectedTeam) {
+      domRef.current.scrollTo(0, 0);
     }
-  }
+  });
```

Comment:
The concept behind `useEffect` is that there is no need to do a compare within the callback function: `if (selectedTeamRef.current !== selectedTeam)`

The proper usage would be to send in `selectedTeam` as a dependency. Then it acts like a `componentDidUpdate` and will compare `if (selectedTeamPrev !== selectedTeam)`

In that case we can simplify our useEffect to this:

```
  const domRef = useRef();

  useEffect(() => {
    domRef.current.scrollTo(0, 0);
  }, [selectedTeam]);
```

notice, that `selectedTeam` is the dependency injected and useEffect is listening to changes once `selectedTeam` changes.

Reply:
Further refined, just use `teamId`:
```
  useEffect(() => {
    domRef.current.scrollTo(0, 0);
  }, [teamId]);
```